fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -92,12 +94,18 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
         val view = binding.root
         historyStore = StoreProvider.get(this) {
             HistoryFragmentStore(
-                HistoryFragmentState(
-                    items = listOf(),
-                    mode = HistoryFragmentState.Mode.Normal,
-                    pendingDeletionItems = emptySet(),
-                    isEmpty = false,
-                    isDeletingItems = false,
+                initialState = HistoryFragmentState.initial,
+                middleware = listOf(
+                    HistoryNavigationMiddleware(
+                        navController = findNavController(),
+                        openToBrowser = ::openItem,
+                    ),
+                    HistoryTelemetryMiddleware(
+                        isInPrivateMode = {
+                            (activity as HomeActivity)
+                                .browsingModeManager.mode == BrowsingMode.Private
+                        },
```

Comment:
you might be able to get away with this being a static boolean vs a lambda, since the whole activity is recreated anyway when toggling between normal/private. Though there also isn't a toggle to do so in the History flow anyway (right?).

Reply:
Ah, that's a good point! I'll change the lambda to just run during instantiation instead